src/test/components/elements/panels/SelectedEventDetails/GenderDetails.test.tsx

```diff
@@ -0,0 +1,47 @@
+import { GenderDetails, countGenders } from "../../../../../app/components/elements/panels/SelectedEventDetails";
+import { asPercentage } from "../../../../../app/services";
+import { mockEventBookings } from "../../../../../mocks/data";
+import { renderTestEnvironment } from "../../../../utils";
+import { screen } from "@testing-library/react";
+
+describe("GenderDetails", () => {
+  const setupTest = (props = {}) => {
+    renderTestEnvironment({
+      PageComponent: GenderDetails,
+      componentProps: {
+        ...props,
+      },
+      initialRouteEntries: ["/admin/events/"],
+    });
+  };
+
+  it("renders a list of 0 count genders when there are no event bookings", () => {
+    setupTest({ eventBookings: [] });
+    const eventGenders = screen.getByTestId("event-genders");
+    expect(eventGenders).toHaveTextContent("Gender:");
+    const genderStats = screen.getByRole("list");
+    const expectedValues = ["Male", "Female", "Other", "Prefer not to say", "Unknown"];
+    expectedValues.forEach((value) => {
+      expect(genderStats).toHaveTextContent(`${value}: 0 (0%)`);
+    });
+  });
+
+  it("renders a list of genders with correct counts when there are event bookings", () => {
```

Comment:
There is a small issue with this test: you've essentially copy-pasted a big chunk of the implementation here (`numberOfConfirmedOrAttendedBookings` is being calculated in the same way as it's done in the implementation, and you're also calling `countGenders`, just as you'd do in the implementation). This is not good for a number of reasons. For instance, you should be focusing on testing the behaviour, rather than the implementation. Also, if you accidentally change the implementation of `countGenders`, this test would still pass, when you'd want it to fail. I also need to make an effort to read this test now, because I need to understand all this extra logic, but tests should be as simple as possible.

An easy improvement would be to get rid of `numberOfConfirmedOrAttendedBookings` and `male`, `female`, etc. And simply set the `expectedValues` manually, as a plain string. This is who you would have had to do it if the test had been written before the implementation anyway. Does this make sense?

Reply:
Updated now and is much simpler